person.rb

```diff
@@ -0,0 +1,31 @@
+require_relative './nameable'
+
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @nameable = nameable.new
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def validate_name
+    @name = @nameable.correct_name
+  end
+
+  private :of_age?
+
+  def can_use_services?
+    of_age? || parent_permission == true
+  end
+end
```

Comment:
The `Person` class should inherit from `Nameable` class and should contain the `correct_name` method. Check the code snippet below:


```ruby
class Person < Nameable
  .
  .

    def correct_name
       name
    end
  .
  .
end
```